tool/src/java/mod.rs

```diff
@@ -0,0 +1,104 @@
+use std::collections::HashSet;
+use std::fs::File;
+use std::io::Write;
+use std::path::Path;
+
+use diplomat_core::hir::TypeContext;
+
+use crate::c2;
+use crate::common::FileMap;
+
+const TMP_C_DIR: &str = "tmp";
+const LIBRARY: &str = "somelib"; // todo: build from conf. Ensure that name is not the same as any
+                                 // type
+const GROUP: &str = "dev.diplomattest"; // todo: config
+const TMP_LIB_NAME: &str = "dev/diplomattest/somelib"; // todo: build from conf
+const JAVA_DIR: &str = "src/main/java/";
+
+pub fn run(
+    tcx: &TypeContext,
+    _conf_path: Option<&Path>,
+    out_folder: &Path,
+) -> std::io::Result<FileMap> {
+    let files = FileMap::default();
+    let mut context = c2::CContext::new(tcx, files, false);
+    context.run();
```

Comment:
Note that after https://github.com/rust-diplomat/diplomat/pull/495, the C2 backend can now be directly invoked per-type in a cleaner fashion, if you decide that's more useful

Reply:
I'd prefer to implement that as a followup.

Reply:
I actually implemented that. Biggest problem is that I need the C runtime too, which I made pub (crate)